### PR TITLE
feat(spanner): support data_boost_enabled for partitioned query and read

### PR DESCRIPTION
**Pull request by NivedhaSenthil**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/ruby-spanner/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [ ] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [ ] Update code documentation if necessary.

closes: #<issue_number_goes_here>

### CODE REVIEW

File: google-cloud-spanner/test/google/cloud/spanner/batch_snapshot/execute_partition_test.rb [hunk 1]
```diff
@@ -260,7 +260,7 @@
       columns: ["id", "name", "active", "age", "score", "updated_at", "birthday", "avatar", "project_ids"],
       key_set: Google::Cloud::Spanner::V1::KeySet.new(all: true),
       transaction: tx_selector, index: "", limit: nil, resume_token: nil, partition_token: partition_token,
-      request_options: nil
+      request_options: nil, data_boost_enabled: false
```
Comment (gauravpurohit06) [inline google-cloud-spanner/test/google/cloud/spanner/batch_snapshot/execute_partition_test.rb:263]: If we have defined `data_boost_enabled ` with default value as `false` in function signature, then can we skip it here?

**Review by gauravpurohit06 — APPROVED** [review]
LGTM with minor comment, Can we also the update the sample or it's part of some different repository?

File: google-cloud-spanner/test/google/cloud/spanner/batch_snapshot/execute_partition_test.rb (hunk 1 above)
Comment (NivedhaSenthil) [reply google-cloud-spanner/test/google/cloud/spanner/batch_snapshot/execute_partition_test.rb:263]: Yup it is defined with default value as false, the change here is for the mock method to expect it with the default value.

**Review by dazuma — APPROVED** [review]
LGTM

**Pull request merged**